src/plugins/intel_gpu/src/plugin/ops/interpolate.cpp

```diff
@@ -49,7 +49,10 @@ static void CreateInterpolateOp(Program& p, const std::shared_ptr<ngraph::op::v4
         }
     }
 
-    OPENVINO_ASSERT(!scales_constant || axes.size() == scales.size(), op->get_friendly_name(), " Incorrect axes and scales should be the same size");
+    bool correct_params = !scales_constant;
+    correct_params |= axes.size() == scales.size();
+    correct_params |= attrs.shape_calculation_mode == ov::op::v4::Interpolate::ShapeCalcMode::SIZES;
```

Comment:
Shouldn't "scales" input be related to the `ShapeCalcMode::SCALES`?
```suggestion
    correct_params |= attrs.shape_calculation_mode == ov::op::v4::Interpolate::ShapeCalcMode::SCALES;
```

Reply:
Yeah, you are right. And my idea was to allow further execution in two scenarios: either `axes.size() == scales.size();` and we can verify it (scales_constant exists) or current mode is `ShapeCalcMode::SIZES`, and we can ignore scales related conditions.
Maybe it is better to rewrite the condition in the following way:
```
    if (attrs.shape_calculation_mode == ov::op::v4::Interpolate::ShapeCalcMode::SCALES && scales_constant) {
        OPENVINO_ASSERT(axes.size() == scales.size(), "[GPU] Incorrect axes and scales values for Interpolate operation with id ", op->get_friendly_name());
    }
```
Since this `axes.size() == scales.size()` condition should be checked in case of ShapeCalcMode::SCALES mode and if scales constants are available. What do you think? (Both non-constant sizes and scales are supported by GPU Plugin)